client/src/connection/itc/script.ts

```diff
@@ -224,5 +228,205 @@ class SASRunner{
 
     Write-Host "${LineCodes.ResultsFetchedCode}"
   }
+
+  [void]DeleteItemAtPath([string]$filePath,[bool]$recursive) {
+    if ($recursive) {
+      $items = $this.GetItemsAtPath($filePath);
+      for($i = 0; $i -lt $items.Count; $i++) {
+        if ($items[$i].category -eq 0) {
+          $this.DeleteItemAtPath($items[$i].uri, $true);
+        } else {
+          $this.DeleteItemAtPath($items[$i].uri, $false);
+        }
+      }
+      $this.objSAS.FileService.DeleteFile($filePath)
+    } else {
+      $this.objSAS.FileService.DeleteFile($filePath)
+    }
+  }
+
+  [void]DeleteFile([string]$filePath) {
+    $recursive = $true
+    try {
+      # If we error out, that means we're trying to get items at a file path,
+      # which isn't valid (thus, this isn't recursive)
+      $items = $this.GetItemsAtPath($filePath);
+    } catch {
+      $recursive = $false
+    }
+    try {
+      $this.DeleteItemAtPath($filePath, $recursive)
+      Write-Host (@{success=$true} | ConvertTo-Json)
+    } catch {
+      Write-Host (@{success=$false; message=$Error[0].Exception.Message} | ConvertTo-Json)
+    }
+  }
+
+  [void]CreateDirectory([string]$folderPath, [string]$folderName) {
+    try {
+      $currentFolder = $this.GetItemAtPathWithName($folderPath, $folderName)
+      if ($currentFolder -ne $null) {
+        Write-Host (@{success=$false} | ConvertTo-Json)
+        return;
+      }
+
+      $uri = $this.objSAS.FileService.MakeDirectory($folderPath, $folderName)
+      $newFolder = $this.GetItemAtPathWithName($folderPath, $folderName)
+      Write-Host (@{success=$true; data=$newFolder} | ConvertTo-Json)
+    } catch {
+      Write-Host (@{success=$false; message=$Error[0].Exception.Message} | ConvertTo-Json)
```

Comment:
I don't see code to display this error message to user, do we need to? or at least console.dir?

Reply:
We display some more "generic" messages to the user when things fail, but don't provide the extra detail. I think using `console.dir` could be helpful here. I'll take a stab at that today.